trlx/trainer/accelerate_base_trainer.py

```diff
@@ -467,78 +481,89 @@ def learn(self):  # noqa: C901
         # For each epoch
         for _ in range(self.config.train.epochs):
             # For each batch
-            for batch in self.train_dataloader:
-                # For each update per batch
+            dataloader_iterator = iter(self.train_dataloader)
+            for initial_batch in dataloader_iterator:
+                batches = [initial_batch]
+                if self.accelerator.gradient_accumulation_steps > 1 and self.n_updates_per_batch > 1:
```

Comment:
this check could be skipped but if someone was using a custom dynamic dataloader for rollouts in ppo, it would change the behavior to load up more batches in memory. I don't know how far we want to go to support those cases though. 

and ofc it still changes the behavior if they were using rgadient_accumulation_steps > 1, but I assume nobody was doing that for n_updates_per_batch > 1 since I don't think there's a point in doing that (accumulating gradients over the same batch multiple times)